framework/src/runtime.rs

```diff
@@ -49,21 +56,35 @@ impl Runtime {
     where
         T: Fn(i32) -> std::result::Result<(), i32> + 'static,
     {
-        self.on_signal = Box::new(on_signal);
+        self.on_signal = Arc::new(on_signal);
+    }
+
+    /// Adds a repeated task to run at an interval
+    pub fn add_task_to_run<T>(&mut self, task: T, interval: Duration)
```

Comment:
@drunkirishcoder do we also want to allow for a task that is not on an interval, like a one off, or run via a timer just once (like run some task 5 seconds from now)? I was just (re?)-reading https://tokio.rs/docs/going-deeper/timers/ and thinking about how some variations may be useful.

Reply:
Can't think of a use case for a run-once behavior right now. The run once tasks we do are usually initialization things that don't need to be scheduled. But if we do encounter an use case, the capability can be pretty easily added to the runtime later.

Reply:
I think most scheduling systems I've seen have these variations, scheduling a single task after a delay, or scheduling a repeated task after a delay (and with a delay between them). That said, agree with @drunkirishcoder... if we don't need it right now, I think we probably hold off until we do.

Reply:
Yep agreed. Also I may have lied about it being easy. I can see more likely you would want the ability to schedule a run-once task from inside a pipeline and there would be some threading issues to work through. nonetheless we will get to it when it comes up.